### PR TITLE
HV: correct mmap info for multiboot2

### PR DESCRIPTION
**Pull request by jsun26intel**
The acrn_mbi.mi_mmap_va should point to struct multiboot2_mmap_entry when
boot from multiboot2, which is different from struct multiboot_mmap when
boot from multiboot1. So we should handle mmap info separately for multiboot2.

Tracked-On: #4885

Signed-off-by: Victor Sun <victor.sun@intel.com>

### CODE REVIEW

**Comment by HuiHuangShi** [comment]
4 new potential violations to the coding guideline detected. Please help check. Thanks!<br><1> config:common filename:/hypervisor/boot/multiboot.c func:sanitize_multiboot_info offset:60<br> reason:Casting operation to a pointer. : (void* to struct multiboot2_mmap_entry*): ( struct multiboot2_mmap_entry * ) acrn_mbi.mi_mmap_va<br><2> config:common filename:/hypervisor/boot/multiboot.c func:sanitize_multiboot_info offset:60<br> reason:Cast on a constant value.                    : acrn_mbi.mi_mmap_va<br><3> config:common filename:/hypervisor/boot/multiboot.c func:sanitize_multiboot_info offset:60<br> reason:Casting operation on a pointer. : (void* to struct multiboot2_mmap_entry*): ( struct multiboot2_mmap_entry * ) acrn_mbi.mi_mmap_va<br><4> config:common filename:/hypervisor/boot/multiboot.c func:sanitize_multiboot_info offset:66<br> reason:No cast for widening complex int expression (MR). :  (unsigned int and unsigned char): acrn_mbi.mi_mmap_entry [ i ].type = ( mb2_mmap + i ) -> type<br><br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

File: hypervisor/boot/multiboot.c [hunk 1]
```diff
@@ -41,29 +41,34 @@ void init_acrn_multiboot_info(void)
...
+			uint32_t i;
+			struct multiboot2_mmap_entry *mb2_mmap = (struct multiboot2_mmap_entry *)acrn_mbi.mi_mmap_va;
+
+			pr_info("Multiboot2 detected.");
+			for (i = 0U; i < acrn_mbi.mi_mmap_entries; i++) {
+				acrn_mbi.mi_mmap_entry[i].baseaddr = (mb2_mmap + i)->addr;
+				acrn_mbi.mi_mmap_entry[i].length = (mb2_mmap + i)->len;
+				acrn_mbi.mi_mmap_entry[i].type = (mb2_mmap + i)->type;
+			}
+#endif
```
Comment (dongyaozu) [inline hypervisor/boot/multiboot.c:68]: Given that boot_from_multiboot2 & boot_from_multiboot is very lightweight. How about we do this way:
if (boot_from_multiboot()) {
....
}
if (boot_from_multiboot2()) {
...
}

This way, we save the last else {..}.

File: hypervisor/boot/multiboot.c [hunk 2]
```diff
@@ -41,29 +41,34 @@ void init_acrn_multiboot_info(void)
...
 			acrn_mbi.mi_mmap_entries = E820_MAX_ENTRIES;
 		}
-		(void)memcpy_s((void *)(&acrn_mbi.mi_mmap_entry[0]),
-			(acrn_mbi.mi_mmap_entries * mmap_entry_size),
-			(const void *)acrn_mbi.mi_mmap_va,
-			(acrn_mbi.mi_mmap_entries * mmap_entry_size));
+		if (boot_from_multiboot1()) {
+			uint32_t mmap_entry_size = sizeof(struct multiboot_mmap);
+
+			pr_info("Multiboot1 detected.");
```
Comment (dongyaozu) [inline hypervisor/boot/multiboot.c:52]: do we need seperate pr_info? 1:this is very easy to know from boot cmd line, 2: even if we want to print, we can print boot_regs[0]， or we can print all boot_regs somewhere.

**Comment by HuiHuangShi** [comment]
4 new potential violations to the coding guideline detected. Please help check. Thanks!<br><1> config:common filename:/hypervisor/boot/multiboot.c func:sanitize_multiboot_info offset:60<br> reason:Casting operation to a pointer. : (void* to struct multiboot2_mmap_entry*): ( struct multiboot2_mmap_entry * ) acrn_mbi.mi_mmap_va<br><2> config:common filename:/hypervisor/boot/multiboot.c func:sanitize_multiboot_info offset:60<br> reason:Cast on a constant value.                    : acrn_mbi.mi_mmap_va<br><3> config:common filename:/hypervisor/boot/multiboot.c func:sanitize_multiboot_info offset:60<br> reason:Casting operation on a pointer. : (void* to struct multiboot2_mmap_entry*): ( struct multiboot2_mmap_entry * ) acrn_mbi.mi_mmap_va<br><4> config:common filename:/hypervisor/boot/multiboot.c func:sanitize_multiboot_info offset:65<br> reason:No cast for widening complex int expression (MR). :  (unsigned int and unsigned char): acrn_mbi.mi_mmap_entry [ i ].type = ( mb2_mmap + i ) -> type<br><br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

File: hypervisor/boot/multiboot.c (hunk 2 above)
Comment (jsun26intel) [reply hypervisor/boot/multiboot.c]: @dongyaozu 
updated, please check.

**Pull request merged**